code/ui/inputbox.cpp

```diff
@@ -80,6 +80,30 @@ void UI_INPUTBOX::init_cursor()
 
 void UI_INPUTBOX::create(UI_WINDOW *wnd, int _x, int _y, int _w, int _text_len, const char *_text, int _flags, int pixel_lim, color *clr)
 {
+	textListener = os::events::addEventListener(SDL_TEXTINPUT, 10000, [this](const SDL_Event& event) {
+		bool focus = my_wnd->selected_gadget == this;
```

Comment:
I'm generally not a fan of lambdas used in this fashion, I prefer instead making this a member function or a local helper function.  Will defer to others opinions, however.

Reply:
uhh, sure. I personally think lambdas are quite neat, especially with capturing this, but I can make it a member function (which would need a bind though). Local helper is annoying IMO, since we do need to refer to this fairly often